webpack.dev.config.js

```diff
@@ -0,0 +1,116 @@
+const HtmlWebpackPlugin = require('html-webpack-plugin')
+const FaviconsWebpackPlugin = require('favicons-webpack-plugin')
+
+const path = require('path')
+const webpack = require('webpack')
+const pkg = require('./package.json')
+
+const nodeEnv = process.env.NODE_ENV || 'development'
+const version = process.env.BUILD_VERSION || pkg.version
+const build = process.env.BUILD_NUMBER || 'SNAPSHOT'
+
+const config = require('./src/config.json')
+
+// const ethereumHost = process.env.ETHEREUM_HOST
+const gnosisDbUrl =
+  process.env.GNOSISDB_HOST || `${config.gnosisdb.protocol}://${config.gnosisdb.host}:${config.gnosisdb.port}`
+
+module.exports = {
+  context: path.join(__dirname, 'src'),
+  entry: ['react-hot-loader/patch', 'bootstrap-loader', 'index.js'],
+  devtool: 'cheap-eval-source-map',
+  output: {
+    path: `${__dirname}/dist`,
+    filename: 'bundle.js',
+  },
+  resolve: {
+    symlinks: false,
+    modules: [
+      `${__dirname}/src`,
+      `${__dirname}/package.json`,
+      'node_modules',
+      `${__dirname}/../gnosis.js`,
+      `${__dirname}/../gnosis.js/node_modules`,
+    ],
+  },
+  module: {
+    rules: [
+      {
+        test: /\.(js|jsx)$/,
+        exclude: /(node_modules)/,
+        use: 'babel-loader',
+      },
+      {
+        test: /\.(jpe?g|png|svg)$/i,
+        loader: 'file-loader?hash=sha512&digest=hex&name=img/[hash].[ext]',
+      },
+      {
+        test: /\.(less|css)$/,
+        use: [
+          {
+            loader: 'style-loader',
+          },
+          {
+            loader: 'css-loader',
+          },
+          {
+            loader: 'postcss-loader',
+            options: {
+              sourceMap: true,
+            },
+          },
+          { loader: 'less-loader', options: { strictMath: true } },
+        ],
+      },
+      {
+        test: /\.(ttf|otf|eot|woff(2)?)(\?[a-z0-9]+)?$/,
+        loader: 'file-loader?name=fonts/[name].[ext]',
+      },
+    ],
+  },
+  devServer: {
+    disableHostCheck: true,
+    contentBase: false,
+    port: 5000,
+    proxy: {
+      '/api': {
+        target: gnosisDbUrl,
+        secure: false,
+      },
+    },
+    watchOptions: {
+      ignored: /node_modules/,
```

Comment:
Often enough I have to link my local gnosis.js package and when I compile contracts from there, it should rebuild. Can we remove this, or is this a huge deal for performance/bundlesize?

Reply:
@andre-meyer Its done for webpack-dev-server so it doesn't watch for changes in node_modules folder. I had problems with react-hot-loader not working because node_modules is large and to avoid checking all node_modules large tree it decides to recompile the app instead of HMR

Reply:
Alright! 👍